app/Http/Controllers/Auth/SocialLoginController.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace App\Http\Controllers\Auth;
+
+use Auth;
+use App\Http\Controllers\Controller;
+use App\Http\Requests;
+use App\User;
+use Illuminate\Http\Request;
+use Socialite;
+
+class SocialLoginController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware(['social', 'guest']);
```

Comment:
Let's move this middleware declaration out of the controller and into the routes file (in a route group or something if it effects multiple routes).